src/rgw/rgw_realm_reloader.cc

```diff
@@ -183,7 +183,10 @@ void RGWRealmReloader::reload()
    * the dynamic reconfiguration. */
   env.auth_registry = rgw::auth::StrategyRegistry::create(
       cct, implicit_tenants, env.driver);
-  env.lua.manager = env.driver->get_lua_manager();
+  env.lua.manager = env.driver->get_lua_manager(env.lua.manager->luarocks_path());
+  if (env.lua.background) {
+    env.lua.background->set_manager(env.lua.manager.get());
```

Comment:
the lua background thread will hold a dangling pointer to the old `env.lua.manager` between lines 186 and 188, and nothing prevents it from trying to access that in the meantime

p.s. what happens if realm reload deletes the old LuaManager while it's in the middle of `RadosLuaManager::reload_packages()`?